docs/docs/reference/fast-refresh.md

```diff
@@ -0,0 +1,55 @@
+---
+title: Fast Refresh
+---
+
+Fast Refresh is an implementation of Hot Reloading with full support from React and it replaces purely userland solutions like [`react-hot-loader`](https://github.com/gaearon/react-hot-loader). It is a feature that allows you to get direct feedback for changes in your React components: A change in your editor immediately updates the browser view and if you have errors in your code the browser view displays an error overlay. Fast Refresh is enabled by default in Gatsby's development server and should show your changes within a second, without losing component state which results in an overall better development experience (DX).
```

Comment:
Thanks for updating this, this is helpful context!

Made a few small tweaks: moved some sentences around and added a link to the "react-refresh" package (although maybe it's better to swap that out for [Dan's comment](https://github.com/facebook/react/issues/16604#issuecomment-528663101) in that PR?).

I haven't heard the term "userland" before, what does it mean? Third party? Community created? (I swapped in "unofficial", but feel free to pick another adjective.)

```suggestion
[Fast Refresh](https://www.npmjs.com/package/react-refresh) is an implementation of Hot Reloading with full support from React. It replaces unofficial solutions like [`react-hot-loader`](https://github.com/gaearon/react-hot-loader).

With Fast Refresh, changes to the code for your React components immediately update in the browser, without losing component state. If you have errors in your code, the browser view displays an error overlay with more details about what went wrong. This direct feedback on your changes results in a better overall development experience (DX).

Fast Refresh is enabled by default in Gatsby's development server and should show your changes within a second.
```